GitHub Workflow, `publish-to-npm` failing

### 💡 Description

The `publish-to-npm` GitHub Workflow failed after creating our initial release with the error:

```console
...
npm error code ENEEDAUTH
npm error need auth This command requires you to be logged in to https://registry.npmjs.org/
npm error need auth You need to authorize this machine using `npm adduser`
npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2024-10-23T16_45_32_244Z-debug-0.log
Error: Process completed with exit code 1.
```

Full execution log: https://github.com/NASA-PDS/wds/actions/runs/11484423314/job/31962160635#step:7:131

### ⚔️ Parent Epic / Related Tickets

_No response_

I updated the publish to npm workflow where I updated the NPMJS Token variable name to match what we have stored in our GH Org's secrets. I created a "dev" release against my working branch to test it. The fix appears to have fixed the issue reported in this ticket; however, we're now experiencing a new issue 

```
...
npm error code E404
npm error 404 Not Found - PUT https://registry.npmjs.org/@nasapds%2fwds - Not found
npm error 404
npm error 404  '@nasapds/wds@0.0.1' is not in this registry.
npm error 404
npm error 404 Note that you can also install from a
npm error 404 tarball, folder, http url, or git url.
npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2024-10-30T21_10_50_940Z-debug-0.log
Error: Process completed with exit code 1.
```

Reference GitHub Action Execution Log:  https://github.com/NASA-PDS/wds/actions/runs/11601706554/job/32305071426#step:7:135

I will continue to investigate the issue.

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -22,6 +22,8 @@ jobs:
         run: npm run build:icons
       - name: Build CSS
         run: npm run build:css
+      - name: Login
+        run: npm login
       - name: Publish
         run: npm publish --verbose
         continue-on-error: true

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -23,7 +23,7 @@ jobs:
       - name: Build CSS
         run: npm run build:css
       - name: Publish
-        run: npm publish
+        run: npm publish --verbose
         continue-on-error: true
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPMJS_COM_TOKEN }}

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -19,6 +19,7 @@ jobs:
           cache: npm
           node-version: 'lts/*'
           registry-url: 'https://registry.npmjs.org'
+          scope: '@nasapds'
       - name: Install Dependencies
         run: npm ci
       - name: Build Icons

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -25,7 +25,7 @@ jobs:
       - name: Login
         run: npm login
       - name: Publish
-        run: npm publish --verbose
+        run: npm publish --verbose --access public
         continue-on-error: true
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPMJS_COM_TOKEN }}

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -26,4 +26,4 @@ jobs:
         run: npm publish
         continue-on-error: true
         env:
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPMJS_COM_TOKEN }}

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -9,6 +9,9 @@ on:
 jobs:
   publish-to-npm:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
@@ -25,7 +28,7 @@ jobs:
       - name: Login
         run: npm login
       - name: Publish
-        run: npm publish --verbose --access public
+        run: npm publish --verbose --provenance --access public
         continue-on-error: true
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPMJS_COM_TOKEN }}

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -25,8 +25,6 @@ jobs:
         run: npm run build:icons
       - name: Build CSS
         run: npm run build:css
-      - name: Login
-        run: npm login
       - name: Publish
         run: npm publish --verbose --provenance --access public
         continue-on-error: true

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -13,8 +13,8 @@ jobs:
       contents: read
       id-token: write
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
           cache: npm
           node-version: 'lts/*'

--- a/.github/workflows/publish-to-npm.yml
+++ b/.github/workflows/publish-to-npm.yml
@@ -19,7 +19,6 @@ jobs:
           cache: npm
           node-version: 'lts/*'
           registry-url: 'https://registry.npmjs.org'
-          scope: '@nasapds'
       - name: Install Dependencies
         run: npm ci
       - name: Build Icons

--- a/.npmignore
+++ b/.npmignore
@@ -1,2 +1,16 @@
 .DS_Store
+.editorconfig
+.envrc
+.git
+.github
+.gitignore
+.nvmrc
+.prettierignore
+.prettierrc
+.stylelintrc.json
+Makefile
+README.md
+SECURITY.md
+example
 node_modules
+tests

--- a/package.json
+++ b/package.json
@@ -6,7 +6,7 @@
   "license": "Apache-2.0",
   "repository": {
     "type": "git",
-    "url": "https://github.com/NASA-PDS/wds.git"
+    "url": "git+https://github.com/NASA-PDS/wds.git"
   },
   "scripts": {
     "build:css": "rm -rf css && sass scss:css",

